idseq_pipeline/commands/curate_accession2taxid.py

```diff
@@ -0,0 +1,102 @@
+"""The curate_accession2taxid command."""
+from .base import Base
+import sys
+import time
+import subprocess
+import random
+import pickle
+import shelve
+import argparse
+import gzip
+import re
+
+def curate_taxon_dict(nt_file, nr_file, mapping_files, output_mapping_file):
+    """Curate accessiont2taxid mapping based on existence in NT/NR"""
+    print "Read the nt/nr file"
+    # Read the nt/nr file
+    dbids = {}
+    lines = 0
+    for gzf in [nr_file, nt_file]:  
+        with gzip.open(gzf, 'rb') as seqf:
+            for line in seqf:
+                lines += 1
+                if lines % 100000 == 0:
+                    print "%f M lines. " %  (lines/1000000.0)
+                if line[0] == '>': # header line
+                    s = re.match('^>([^ \.]*).*', line)
+                    if s:
+                        dbids[s.group(1)] = 1
+    print "Read the mapping file and select ..."
+    # Read the accession2taxid mapping files
+    outf = open(output_mapping_file, 'wb')
+    lines = 0
+    for mapping_file in mapping_files:
+        with gzip.open(mapping_file, 'rb') as mapf:
+            for line in mapf:
+                fields = line.split("\t")
```

Comment:
If you only use the first 3 fields, performance may be marginally improved by providing a second argument to `split`, like `line.split("\t", 3)`.

Reply:
You are using a dictionary as a set.   I imagine the memory use might be slightly better if you actually use a set.

Reply:
ok